starlette/concurrency.py

```diff
@@ -22,3 +23,24 @@
         # loop.run_in_executor doesn't accept 'kwargs', so bind them in here
         func = functools.partial(func, **kwargs)
     return await loop.run_in_executor(None, func, *args)
+
+
+class _StopSyncIteration(Exception):
+    pass
+
+
+def _interceptable_next(iterator: Iterator) -> Any:
+    try:
+        result = next(iterator)
+        return result
+    except StopIteration:
+        raise _StopSyncIteration
+
+
+async def iterator_to_async(iterator: Iterator) -> AsyncGenerator:
```

Comment:
How about `iterate_in_threadpool` to mirror `run_in_threadpool`.

Reply:
Great idea. I struggled a lot to name it and wasn't quite convinced either.